src/compute/encryption.rs

```diff
@@ -0,0 +1,974 @@
+            fs::read(&encrypted_file_in_dir).expect("Failed to read encrypted file from dir");
+        assert!(!encrypted_content.is_empty());
+        assert_ne!(encrypted_content, original_data);
+
+        let aes_key_file_in_dir = output_dir_path.join("aes-key.rsa");
+        assert!(
+            aes_key_file_in_dir.exists(),
+            "AES key file not found in output directory."
+        );
+
+        let aes_key_content_bytes =
+            fs::read(&aes_key_file_in_dir).expect("Failed to read AES key file from dir");
+        assert!(!aes_key_content_bytes.is_empty());
+        // AES key is now stored as raw encrypted bytes, not Base64 string
+        assert_eq!(aes_key_content_bytes.len(), 256); // RSA-2048 produces 256-byte output
+    }
+
+    #[test]
+    fn encrypt_data_produces_zip_file_when_valid_input_and_produce_zip_true() {
+        let base_temp = tempdir().expect("Failed to create base temp dir for encrypt_data test");
+        let input_dir = base_temp.path().join("input_data_dir");
+        fs::create_dir_all(&input_dir).expect("Failed to create input_data_dir");
+
+        let input_file_path = input_dir.join("my_result_data.txt");
+        let original_data = b"This is the secret data to be encrypted and zipped!";
+        fs::write(&input_file_path, original_data)
+            .expect("Failed to write to temporary input file");
+
+        let result = encrypt_data(
+            input_file_path.to_str().unwrap(),
+            TEST_RSA_PUBLIC_KEY_PEM,
+            true, // produce_zip = true
+        );
+        assert!(
+            result.is_ok(),
+            "encrypt_data should succeed. Error: {:?}",
+            result.err()
+        );
+
+        let output_zip_path_str = result.unwrap();
+        let output_zip_path = Path::new(&output_zip_path_str);
+        assert!(
+            output_zip_path.exists(),
+            "Output zip file should exist at {}",
+            output_zip_path_str
+        );
+        assert_eq!(
+            output_zip_path.extension().unwrap_or_default(),
+            "zip",
+            "Output file should be a zip file"
+        );
+
+        let expected_zip_parent = input_dir; // work_dir
+        let expected_zip_name_stem = "iexec_out";
+        assert_eq!(
+            output_zip_path.parent().unwrap(),
+            expected_zip_parent,
+            "Zip file created in unexpected parent directory."
+        );
+        assert_eq!(
+            output_zip_path.file_stem().unwrap().to_str().unwrap(),
+            expected_zip_name_stem,
+            "Zip file has unexpected stem."
+        );
+
+        let zip_file_reader = File::open(output_zip_path).expect("Failed to open output zip file");
+        let mut archive =
+            ZipArchive::new(zip_file_reader).expect("Failed to read output zip archive");
+        assert!(
+            archive.len() == 2,
+            "Zip archive should contain 2 files. Found: {}",
+            archive.len()
+        );
```

Comment:
```suggestion
        assert_eq!(
            archive.len(),
            2,
            "Zip archive should contain 2 files. Found: {}",
            archive.len()
        );
```
The IDE suggests that I use assert_eq!